src/Extraction/ImageExtractor.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+namespace Mindee\Extraction;
+
+use Mindee\Error\MindeeGeometryException;
+use Mindee\Error\MindeePDFException;
+use Mindee\Error\MindeeUnhandledException;
+use Mindee\Geometry\BBox;
+use Mindee\Geometry\BBoxUtils;
+use Mindee\Input\LocalInputSource;
+use Mindee\Parsing\DependencyChecker;
+use Mindee\Parsing\Standard\BaseField;
+
+/**
+ * Extract sub-images from an image.
+ */
+class ImageExtractor
+{
+    /**
+     * Array of extracted page images.
+     *
+     * @var array
+     */
+    private array $pageImages = [];
+    /**
+     * Name of the file.
+     *
+     * @var string
+     */
+    private string $filename;
+    /**
+     * Format to save the image as.
+     *
+     * @var string
+     */
+    private string $saveFormat;
+    /**
+     * Local input object used by the ImageExtractor.
+     *
+     * @var LocalInputSource
+     */
+    protected LocalInputSource $inputSource;
+
+
+    /**
+     * @param LocalInputSource $localInput Local Input, accepts all compatible formats.
+     * @param string|null      $saveFormat Save format, will be coerced to jpg by default.
+     * @throws MindeeUnhandledException Throws if PDF operations aren't supported.
+     * @throws MindeePDFException Throws if ImageMagick can't handle the image.
+     */
+    public function __construct(LocalInputSource $localInput, ?string $saveFormat = null)
+    {
+        if (!DependencyChecker::isImageMagickAvailable()) {
+            throw new MindeeUnhandledException(
```

Comment:
I would probably just throw the exception in the dep checker to avoid duplicate error message texts.

Reply:
Yeah no clue why I put it there :facepalm: